azurerm/internal/services/advisor/data_source_advisor_recommendations.go

```diff
@@ -0,0 +1,212 @@
+package advisor
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/advisor/mgmt/2020-01-01/advisor"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	uuid "github.com/satori/go.uuid"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+)
+
+func dataSourceArmAdvisorRecommendations() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmAdvisorRecommendationsRead,
+
+		Timeouts: &schema.ResourceTimeout{
+			Read: schema.DefaultTimeout(10 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"categories_filter": {
```

Comment:
Could we change this to
```suggestion
			"filter_by_category": {
```